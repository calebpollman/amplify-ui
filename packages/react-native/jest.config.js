module.exports = {
  preset: 'react-native',
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  collectCoverageFrom: [
    '<rootDir>/src/**/*.{js,jsx,ts,tsx}',
    '!<rootDir>/src/**/*{c,C}onstants.ts',
  ],
  coverageThreshold: {
    global: {
      branches: 16,
      functions: 9,
      lines: 16,
      statements: 16,
    },
  },
};
