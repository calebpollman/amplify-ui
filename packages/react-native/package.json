{
  "name": "@aws-amplify/ui-react-native",
  "version": "0.1.0",
  "main": "dist/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/index.d.ts",
  "license": "Apache-2.0",
  "scripts": {
    "prebuild": "rimraf dist",
    "build": "yarn build:ts && cp -a src/assets dist",
    "build:ts": "ttsc --project tsconfig.json",
    "dev": "yarn build:ts --watch",
    "clean": "rimraf dist",
    "lint": "ttsc --noEmit && eslint src",
    "test": "yarn test:unit",
    "test:watch": "yarn test:unit:watch",
    "test:unit": "jest --config jest.config.js",
    "test:unit:watch": "jest --watch",
    "test:verbose": "yarn test -- --verbose"
  },
  "devDependencies": {
    "@babel/cli": "^7.15.7",
    "@babel/core": "7.15.5",
    "@babel/preset-env": "7.15.6",
    "@testing-library/react-hooks": "^7.0.2",
    "@types/react": "^16.13.1",
    "@types/react-native": "^0.63.45",
    "@types/react-test-renderer": "^17.0.1",
    "@typescript-eslint/eslint-plugin": "^4.32.0",
    "@typescript-eslint/parser": "^4.32.0",
    "@zerollup/ts-transform-paths": "^1.7.18",
    "babel-jest": "^27.2.4",
    "eslint": "^7.32.0",
    "eslint-config-airbnb": "^18.2.1",
    "eslint-config-airbnb-typescript": "^14.0.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-import": "^2.24.2",
    "eslint-plugin-jest": "^24.5.0",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-react": "^7.26.0",
    "eslint-plugin-react-hooks": "^4.2.0",
    "metro-react-native-babel-preset": "^0.66.2",
    "prettier": "2.4.1",
    "react": "latest",
    "react-native": "^0.66.4",
    "react-test-renderer": "^17.0.2",
    "rimraf": "^3.0.2",
    "ts-jest": "^27.1.3",
    "ttypescript": "^1.5.13"
  },
  "dependencies": {
    "@aws-amplify/ui": "3.0.7",
    "@aws-amplify/ui-react-core": "0.1.0",
    "@aws-amplify/notifications": "in-app-messaging"
  },
  "peerDependencies": {
    "aws-amplify": "in-app-messaging",
    "react": ">= 16.8.0",
    "react-native": ">= 0.63.4"
  },
  "resolutions": {
    "@types/react": "^16.13.1"
  }
}
