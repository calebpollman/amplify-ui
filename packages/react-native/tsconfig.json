{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "baseUrl": ".",
    "declaration": true,
    "downlevelIteration": true,
    "esModuleInterop": true,
    "jsx": "react-native",
    "noImplicitAny": false,
    "module": "es2015",
    "moduleResolution": "node",
    "noEmitOnError": true,
    "outDir": "./dist",
    "plugins": [
      {
        "transform": "@zerollup/ts-transform-paths",
        "exclude": ["*"]
      }
    ],
    "paths": { "@ui": ["src/ui"] },
    "skipLibCheck": true,
    "target": "es5"
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"],
  "exclude": ["node_modules", "jest.config.js"]
}
