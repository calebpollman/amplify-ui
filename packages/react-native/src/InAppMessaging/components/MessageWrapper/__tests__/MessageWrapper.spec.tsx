/*
 * Copyright 2017-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Text } from 'react-native';

import MessageWrapper from '../MessageWrapper';

const Children = () => <Text>Children</Text>;

describe('MessageWrapper', () => {
  it('renders as expected', () => {
    const renderer = TestRenderer.create(
      <MessageWrapper>
        <Children />
      </MessageWrapper>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
