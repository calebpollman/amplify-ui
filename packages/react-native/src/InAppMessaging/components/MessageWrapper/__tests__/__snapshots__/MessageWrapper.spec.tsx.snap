// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageWrapper renders as expected 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <Text>
      Children
    </Text>
  </RCTSafeAreaView>
</Modal>
`;
